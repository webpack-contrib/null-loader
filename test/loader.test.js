import {
  getCompiler,
  compile,
  getModuleSource,
  getExecutedCode,
} from './helpers/index';

describe('null-loader', () => {
  it('should works', async () => {
    const compiler = getCompiler('./fixture.js');
    const stats = await compile(compiler);

    expect(getModuleSource('./fixture.js', stats)).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('should works as inline loader', async () => {
    const compiler = getCompiler('./inline-loader.js');
    const stats = await compile(compiler);

    expect(getModuleSource('./inline-loader.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('should works with not js modules', async () => {
    const compiler = getCompiler('./fixture2.js');
    const stats = await compile(compiler);

    expect(getModuleSource('./fixture2.js', stats)).toMatchSnapshot('module');
    expect(getExecutedCode('main.bundle.js', compiler, stats)).toMatchSnapshot(
      'result'
    );
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
