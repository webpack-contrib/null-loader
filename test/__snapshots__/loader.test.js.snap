// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`null-loader should works as inline loader: errors 1`] = `Array []`;

exports[`null-loader should works as inline loader: module 1`] = `""`;

exports[`null-loader should works as inline loader: result 1`] = `Object {}`;

exports[`null-loader should works as inline loader: warnings 1`] = `Array []`;

exports[`null-loader should works with not js modules: errors 1`] = `Array []`;

exports[`null-loader should works with not js modules: module 1`] = `""`;

exports[`null-loader should works with not js modules: result 1`] = `Object {}`;

exports[`null-loader should works with not js modules: warnings 1`] = `Array []`;

exports[`null-loader should works: errors 1`] = `Array []`;

exports[`null-loader should works: module 1`] = `""`;

exports[`null-loader should works: result 1`] = `Object {}`;

exports[`null-loader should works: warnings 1`] = `Array []`;
